lib/src/tenant_token/exceptions.dart

```diff
@@ -0,0 +1,5 @@
+part of tenant_token;
+
+class ExpiredSignatureException implements Exception {}
+
+class InvalidApiKeyException implements Exception {}
```

Comment:
Why we don't put this Exception in the `exception` file that already exists?
More generally, I'm not sure to understand why you create an exception on the structure of this feature (the tenant_tokne) by putting it in a separate folder `src/tenant_token/`. Why not create a file `src/tenant_token_impl.dart` like the others? Do we not stick that the code will be less readable if someone wants to contribute?

Reply:
The following message is a mix to all of your questions 🍣 

If you look carefully, we are doing an anti-pattern here in this sdk, called [God Objects](https://en.wikipedia.org/wiki/God_object).
A God Object is an object who knows everything (that's why God in the name haha).
Quoting wikipedia: "is an object that references a large number of distinct types, has too many unrelated or uncategorized methods, or some combination of both".

So, in the `client_impl` we are doing exactly that and in the `index_impl` as well. 
That's why I didn't created the tenant token direcly in those files.

In this feature I'm trying a new and idiomatic way to make the code less coupled with each other. 
And of course, trying to avoid adding more responsibilities to the (already bloated) client/index classes.

So the main idea was to find a way that I could create a new entity without mixing other domains in it.
Dart has this way to create "libraries" in the project and then call them when you need. Since I created a new "library" for this project I just not make much sense add the new exceptions in other directory than the library one. 

In order to respect this brand new domain (DDD), I shouldn't add the errors to the `exception.dart` and if you give a look into the `exception.dart`, you'll see that only deals with errors caming from the API.

> Why not create a file `src/tenant_token_impl.dart` like the others?

In this SDK we are defining the classes as [anemic models](https://thedomaindrivendesign.io/anemic-model) and making them a interface (but we just use them once), so it could be improved!

In the ruby SDK we have a similar pattern being implemented, we have a lot of different domains in the `Client`, and I tried to start decoupling it by creating a module (https://github.com/meilisearch/meilisearch-ruby/blob/feature/tenant-token/lib/meilisearch/tenant_token.rb). I was able to accomplish what I want there, but is not perfect.

I'm not sure if I could convince you, but if you have any other question please go ahead! 🥇

Reply:
@brunoocasali, Yes, I don't doubt at all that the new implementation should be better on the contrary. I was just wondering about the fact that two different ways of structuring the code co-exist in the SDK now. Maybe it would be good to create an issue about this so that the rest of the SDK is modified in the same way as the new code. This would allow having more coherence in the whole code. What do you think?

Reply:
> Yes, I don't doubt at all that the new implementation should be better on the contrary.

❤️ 

> Maybe it would be good to create an issue about this so that the rest of the SDK is modified in the same way as the new code.

I agree with that, probably it could be the next step, I just want to test this new approach before moving all the code to this new style!

> I was just wondering about the fact that two different ways of structuring the code co-exist in the SDK now.

Yeah, this is not so good in a global POV, but if you look closely this feature is self-contained in the "library", since it is not mixed into the code I think we have a good starting point (should not be harder to the newcomers).